### PR TITLE
UX-2371/EKS CAPI Clusters - Integrate API from Backend to list AMI types for AWSManagedMachinePool

### PR DESCRIPTION
**Pull request by Nalinmujoo**
Screenshot 2022-11-08 at 12 32 08 PM

### CODE REVIEW

**Comment by patil-pratik-87** [comment]
is this still WIP?

File: src/api-client/Sunpike.ts [hunk 1]
```diff
@@ -1523,6 +1524,24 @@ class Sunpike extends ApiService {
       },
     })
   }
+
+  @trackApiMethodMetadata({
+    url: '/sunpike/apis/opt.pf9.io/v1beta1/awsidentities/:identityName/eksamitypes',
+    type: 'GET',
+    params: ['identityName'],
+    // error: { isClusterError: true, k8sResource: 'EKSAMITypeList' },
```
Comment (jruddell) [inline src/api-client/Sunpike.ts:1532]: why is this commented out?

File: src/app/plugins/infrastructure/components/clusters/capi/eks-ami-types/actions.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import ApiClient from 'api-client/ApiClient'
+import Bugsnag from 'utils/bugsnag'
+import ActionsSet from 'core/actions/ActionsSet'
+import ListAction from 'core/actions/ListAction'
+import DataKeys, { entityNamesByKey } from 'k8s/DataKeys'
+
+const { sunpike } = ApiClient.getInstance()
+
+export const eksAmiTypesActions = ActionsSet.make<DataKeys.EksAmiTypes>({
```
Comment (xantorres) [inline src/app/plugins/infrastructure/components/clusters/capi/eks-ami-types/actions.ts:9]: We should avoid creating `ActionSet`'s unnecessarily. If this entity is only be used in a dropdown, creating an independent `ListAction` should be enough, no need to create an `ActionSet` here

File: src/api-client/Sunpike.ts (hunk 1 above)
Comment (Nalinmujoo) [reply src/api-client/Sunpike.ts]: uncommented

File: src/app/plugins/infrastructure/components/clusters/capi/eks-ami-types/actions.ts (hunk 2 above)
Comment (Nalinmujoo) [reply src/app/plugins/infrastructure/components/clusters/capi/eks-ami-types/actions.ts]: changed

**Review by patil-pratik-87 — APPROVED** [review]

**Pull request merged**